staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/metrics_test.go

```diff
@@ -0,0 +1,256 @@
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="0.00256"} 3
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="0.01024"} 3
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="0.04096"} 3
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="0.16384"} 3
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="0.65536"} 3
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="2.62144"} 3
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="Inf"} 3
+        	apiextensions_apiserver_validation_ratcheting_seconds_sum 3.0000000000000004e-09
+        	apiextensions_apiserver_validation_ratcheting_seconds_count 3
+			`,
+		},
+		{
+			desc: "ratcheted change object yields metrics",
+			obj: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"foo": "bar",
+				},
+			},
+			old: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"foo": "barx",
+				},
+			},
+			schema: apiextensions.JSONSchemaProps{
+				Type: "object",
+				Properties: map[string]apiextensions.JSONSchemaProps{
+					"foo": {
+						Type: "string",
+						Enum: []apiextensions.JSON{
+							"incorrect",
+						},
+					},
+				},
+			},
+			want: `
+			# HELP apiextensions_apiserver_validation_ratcheting_seconds [ALPHA] Time for comparison of old to new for the purposes of CRDValidationRatcheting during an UPDATE in seconds.
+        	# TYPE apiextensions_apiserver_validation_ratcheting_seconds histogram
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="1e-05"} 4
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="4e-05"} 4
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="0.00016"} 4
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="0.00064"} 4
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="0.00256"} 4
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="0.01024"} 4
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="0.04096"} 4
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="0.16384"} 4
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="0.65536"} 4
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="2.62144"} 4
+        	apiextensions_apiserver_validation_ratcheting_seconds_bucket{le="Inf"} 4
+        	apiextensions_apiserver_validation_ratcheting_seconds_sum 4.0000000000000004e-09
+        	apiextensions_apiserver_validation_ratcheting_seconds_count 4
+			`,
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.desc, func(t *testing.T) {
+			defer func() {
+				// No reset function for prometheus histogram metric
+				// so recreate it (can't use newValidationMetrics() because it's private
```

Comment:
/hold

cc @logicalhan had some difficulties here resetting the histogram metric since it is not a `Vec` and does not expose a `Reset()`. For now the tests are order dependent. 

Do you have any advice to workaround?